Source/Core/Core/IPC_HLE/WII_IPC_HLE_Device_DI.cpp

```diff
@@ -16,31 +16,25 @@
 #include "Core/IPC_HLE/WII_IPC_HLE.h"
 #include "Core/IPC_HLE/WII_IPC_HLE_Device_DI.h"
 
-static CWII_IPC_HLE_Device_di* g_di_pointer;
 static int ioctl_callback;
 
 static void IOCtlCallback(u64 userdata, int cycles_late)
 {
-	if (g_di_pointer != nullptr)
-		g_di_pointer->FinishIOCtl((DVDInterface::DIInterruptType)userdata);
+	std::shared_ptr<IWII_IPC_HLE_Device> di = WII_IPC_HLE_Interface::GetDeviceByName("/dev/di");
+	if (di)
+		std::static_pointer_cast<CWII_IPC_HLE_Device_di>(di)->FinishIOCtl((DVDInterface::DIInterruptType)userdata);
```

Comment:
Its usually asking for problems if you rely on implementation details of a particular interface, and might hint a place where the interface should be extended. But then again, its nothing that will leave Dolphin (or come from the outside), so its probably fine.